### PR TITLE
feat #106157: Set up details view for vertical SLD items

### PR DESCRIPTION
**Pull request by ChrisSkj**
No colors yet, waiting for Robim

image

### CODE REVIEW

File: src/layers/external/leakage/ui/detailsViewVertical/DetailsViewVertical.module.css [hunk 1]
```diff
@@ -37,6 +38,7 @@
 }
 
 .content {
+  padding: 16px 24px;
```
Comment (ErlendAHall) [inline src/layers/external/leakage/ui/detailsViewVertical/DetailsViewVertical.module.css:41]: the padding can be relative sized
`padding; 1rem 2rem`

File: src/layers/external/leakage/ui/detailsViewVertical/VerticalItems.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    );
+}
+
+interface VerticalItemsProps {
+    items: ExternalCommonLayerFeatureItem<SubseaLeakageMetaData>[];
+}
+
+function VerticalItems(props: VerticalItemsProps) {
+    // do all items have the same detailsUri?
+    const detailsUri = props.items[0].detailsUri;
```
Comment (ErlendAHall) [inline src/layers/external/leakage/ui/detailsViewVertical/VerticalItems.tsx:102]: Should create handling if props.item is 0 in length

File: src/layers/external/leakage/ui/detailsViewVertical/DetailsViewVertical.module.css (hunk 1 above)
Comment (Tobzor) [inline src/layers/external/leakage/ui/detailsViewVertical/DetailsViewVertical.module.css:41]: hmm, this should be relative sizes, no?

File: src/layers/external/leakage/ui/SubseaLeakageHelper.tsx [hunk 3]
```diff
@@ -280,3 +283,19 @@ export function getLeakageData(data: SubseaLeakageCollectionData) {
...
+export type Row = {
+    cells: Cell[];
+};
+
+export type Cell = {
+    columnKey: string;
+    value: string;
+    icon: unknown;
+    link: string;
+};
```
[the remark this answers is not in the document]
Comment (ChrisSkj) [reply src/layers/external/leakage/ui/SubseaLeakageHelper.tsx:301]: Probably. We can move them to common when they are used somewhere else

**Review by Tobzor — APPROVED** [review]

File: src/layers/external/leakage/ui/detailsViewVertical/VerticalItems.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    if (!value) value = '-';
+
+    return (
+        <div>
+            <div className={styles.columnLabel}>{label}</div>
+            <div>{value}</div>
+        </div>
+    );
+}
```
[the remark this answers is not in the document]
Comment (ChrisSkj) [reply src/layers/external/leakage/ui/detailsViewVertical/VerticalItems.tsx:42]: We'll come back to this

**Pull request merged**